src/mongocxx/private/libbson.cpp

```diff
@@ -40,22 +40,22 @@ scoped_bson_t::scoped_bson_t(bsoncxx::document::view_or_value doc)
 scoped_bson_t::scoped_bson_t(bsoncxx::stdx::optional<bsoncxx::document::view_or_value> doc)
     : _is_initialized{doc} {
     if (doc) {
-        _doc = doc;
+        _doc = std::move(doc);
         doc_to_bson_t(*_doc, &_bson);
     }
 }
 
 void scoped_bson_t::init_from_static(bsoncxx::document::view_or_value doc) {
     _is_initialized = true;
-    _doc = doc;
+    _doc = std::move(doc);
     doc_to_bson_t(*_doc, &_bson);
 }
 
 void scoped_bson_t::init_from_static(
     bsoncxx::stdx::optional<bsoncxx::document::view_or_value> doc) {
     if (doc) {
         _is_initialized = true;
-        _doc = doc;
+        _doc = std::move(doc);
```

Comment:
Could this be related to https://jira.mongodb.org/browse/CXX-1388 ?

Reply:
Don't think so, no. This was purely a missed optimization.